pkg/kubelet/pluginmanager/cache/actual_state_of_world.go

```diff
@@ -53,9 +54,13 @@ type ActualStateOfWorld interface {
 	// If a plugin does not exist with the given socket path, this is a no-op.
 	RemovePlugin(socketPath string)
 
-	// PluginExists checks if the given plugin exists in the current actual
+	// PluginExistsWithCorrectTimestamp checks if the given plugin exists in the current actual
```

Comment:
Done

Reply:
Why do we want to deprecate internal APIs? We can just remove it as it's not used anywhere. Am I missing something?

Reply:
This was suggested during the SIG node meting on January 31st (https://docs.google.com/document/d/1Ne57gvidMEWXR70OxxnRkYquAoMpt56o75oZtg-OeBg/edit#):

```
[claudiubelu] Proposed changes to how kubelet detects updates for registered plugins b/c current implementation doesn’t work on Windows due to timestamp granularity issues https://github.com/kubernetes/kubernetes/pull/114136 
https://cs.k8s.io/?q=PluginExistsWithCorrectTimestamp&i=nope&files=&excludeFiles=&repos=
https://testgrid.k8s.io/sig-windows-signal#windows-unit-master test failure caused by timestamp
this has been proposed and brought up in the past, but no reviews came in.
it is one of the last consistently failing tests in the testgrid above.
[MikeBrown] consider keeping existing public method and add another one, deprecate first one later. nod just to protect custom plugin managers test buckets etc .. deprecate first thx.. see:
[https://cs.k8s.io/?q=PluginExistsWithCorrectTimestamp&i=nope&files=&excludeFiles=&repos=](https://cs.k8s.io/?q&#61;PluginExistsWithCorrectTimestamp&i&#61;nope&files&#61;&excludeFiles&#61;&repos&#61;)
```

Reply:
@claudiubelu ok, thanks. So, this API is used somewhere else, it seems. Can you point me out to where it's used?

Reply:
@mikebrow can you help? ^^^^

Reply:
I cannot find good examples either. I'd be ok to keep it as deprecated for now just in case and maybe remove in couple of versions.

Reply:
I'd suggest to either remove it or deprecate **and** create an issue. Otherwise it has big chances to be forgotten.

Reply:
> I'd suggest to either remove it or deprecate **and** create an issue. Otherwise it has big chances to be forgotten.

Created issue: https://github.com/kubernetes/kubernetes/issues/115897